sphinx/writers/latex.py

```diff
@@ -1163,7 +1163,8 @@ def visit_paragraph(self, node: Element) -> None:
         else:
             # the \sphinxAtStartPar is to allow hyphenation of first word of
             # a paragraph in narrow contexts such as in a table cell
-            self.body.append('\n\\sphinxAtStartPar\n')
+            # added as two items (cf. line trimming in depart_entry())
+            self.body.extend(['\n', '\\sphinxAtStartPar\n'])
```

Comment:
are we sure paragraph first child node will never insert an extra ``\n`` after this? In case of doubt:
- we could replace the `\n` after `\\sphinxAtStartPar` by  a space. Then however `\sphinxAtStartPar<space>` will start most paragraphs in latex output rather than standing on a line of its own. This makes absolutely no difference to LaTeX but is less pleasant for simple difference of output with Sphinx before 3.5. 
- or, the first child node should be trimmed for leading `'\n'` (and if nothing is left of this node, then the next one...)

Reply:
As I commented in another PR, all kinds of children for paragraph nodes are inline elements (basically). So no child will insert a blank line before itself. At a glance, it seems visitors in LaTeX writer that inserts `\n` before their macros are block level (vertical mode) elements.